pkg/sdk/go/sdk.go

```diff
@@ -353,3 +358,29 @@ func (sdk mfSDK) sendThingRequest(req *http.Request, key, contentType string) (*
 
 	return sdk.client.Do(req)
 }
+
+func (sdk mfSDK) parseQueryParameters(baseURL, endpoint string, pm PageMetadata) (string, error) {
+	var resp map[string]interface{}
+	var queryParams string
+	jsonMarshal, err := json.Marshal(pm)
+	if err != nil {
+		return "", err
+	}
+	err = json.Unmarshal(jsonMarshal, &resp)
+	if err != nil {
+		return "", err
+	}
+	for key, val := range resp {
+		switch key {
+		case "total", "offset", "limit", "name", "email":
+			queryParams = fmt.Sprintf("%s%s=%s&", queryParams, key, fmt.Sprint(val))
+		case "metadata":
+			metadataJSON, err := json.Marshal(pm.Metadata)
+			if err != nil {
+				return "", err
+			}
+			queryParams = fmt.Sprintf("%s%s=%s&", queryParams, key, string(metadataJSON))
+		}
+	}
```

Comment:
You can attach this to a utility method of PageMetadata that will return URL-encoded query params for the given Page metadata, something like:
```go
func (sdk mfSDK) withQueryParams(baseURL, endpoint string, pm PageMetadata) (string, error) {
    q, err := pm.Query()
    if err != nil {
        return "", err
    }
	return fmt.Sprintf("%s/%s?%s", baseURL, endpoint, q), nil
}
```
where `pm.Query` converts page metadata to query params.